### PR TITLE
add happy eyeballs

### PR DESCRIPTION
**Pull request by zwhitchcox**
patch all to use 'happy-eyeballs' algorithm

fixes #1804 #1787

Change-type: patch

### CODE REVIEW

**Comment by pipex** [comment]
What is the `npmrc` for?

**Comment by zwhitchcox** [comment]
@pipex it was giving me errors when I was trying to build and it switched the package-lock.json to version 2..

Now that I think about it though, I was using version 16.8.0 when I did that, so probably it's not a problem when I'm using 12, so I will take that out, my mistake.

I was wondering why you guys didn't have that!

**Comment by pipex** [comment]
@zwhitchcox are you able to test this on a device? I would like to make sure everything still works. I'll test on one of my devices but I don't have ipv6 support 

If you have a device with a dev mode OS here is how you can test the supervisor https://github.com/balena-os/balena-supervisor#sync

**Comment by zwhitchcox** [comment]
@pipex, found a few compatibility issues with the (deprecated) `request` library, but they are all fixed now.

Good that you suggested testing that!

**Comment by 20k-ultra** [comment]
I am waiting for test this with https://github.com/balena-io-playground/ipv6-router

**Comment by 20k-ultra** [comment]
@balena-ci rebase

**Comment by 20k-ultra** [comment]
I've tested this PR and it does work on a device with IPv6 address but not routing IPv6 traffic. Rebase master and we can approve!

File: package.json [hunk 1]
```diff
@@ -29,6 +29,7 @@
   "private": true,
   "dependencies": {
     "dbus": "^1.0.7",
+    "happy-eyeballs": "0.0.15",
```
Comment (20k-ultra) [inline package.json:32]: can we make this @balena/happy-eyeballs ?

**Comment by 20k-ultra** [comment]
I've removed the merge commit and rebased master. I just need you to look into using @balena instead of the personal repo package.

**Comment by ab77** [comment]
@balena-ci retest

**Comment by 20k-ultra** [comment]
@balena-ci rebase

**Comment by dakshshah96** [comment]
@zwhitchcox Would love to see this land, especially because it's the issue we were (and are) facing which prompted this PR in the first place. Any approximate timeline? No pressure!

**Comment by zwhitchcox** [comment]
@dakshshah96 Thank you for your interest, the module is ready, still working on getting the CI to publish it. For some reason, it's not able to pull the repo. I've pinged the team and hopefully it will get merged tomorrow. That's the goal anyway.

**Comment by pipex** [comment]
@zwhitchcox once you figure out the issue and get jobs to pass, could you make sure to squash your commits before merging?

**Comment by zwhitchcox** [comment]
Yes, @pipex!

**Comment by dakshshah96** [comment]
I'm sorry to be a bother here, but this PR might fix the issue that has been bothering us on balenaCloud for months now. We've had no luck even after changing SIM providers. Any update on when we can expect this to be merged?

**Comment by pipex** [comment]
LGTM!

**Comment by pipex** [comment]
@balena-ci rebase

**Review by pipex — APPROVED** [review]
LGTM

**Comment by pipex** [comment]
@dakshshah96 The PR is merged. Sorry about the delay. Would you like to update your supervisor to 12.10.14 and tell us if that fixes the issue?

**Comment by dakshshah96** [comment]
@pipex No problem at all, thanks for merging this! I'll update few of our devices to the latest version and circle back here in a few days. Cheers!

**Pull request merged**